### ISSUE TITLE
more sort work

### ISSUE THREAD

**Issue by sjneph** [bedops/bedops#163]
For v2p4p21, need to also update mergeSort( ) for sort-bed, and bedops' --union option and the bed_check_iterator used by bedmap/bedextract/checkSort to adhere to the new sorting rules.  Alex is working on the same in starchcat (good catch, Alex!).

**Comment by alexpreynolds** [comment]
In addition to investigating `starchcat`, here is a reminder to test sort order of input to `starch`, e.g.:

```
$ echo -e 'chr1\t100\t120\nchr1\t100\t110' | starch - > /tmp/test.starch
$ echo -e 'chr1\t100\t120\nchr1\t50\t120' | starch - > /tmp/test.starch
$ echo -e 'chr2\t100\t120\nchr1\t100\t110' | starch - > /tmp/test.starch
```

I need to fix any error cases that are not caught.

**Comment by alexpreynolds** [comment]
The first two tests (interval coordinate ordering) are already in `starch`. Commit 502726e0b189c76c1fa2fdf3379ab3da22493f85 includes the third test (interval chromosome name ordering).

**Comment by alexpreynolds** [comment]
Putting in a comment to remind myself to investigate an order test for the BED "remainder" (four or more columns) in `starch`.

**Comment by alexpreynolds** [comment]
Putting in a comment to remind myself to review notes in `STARCHCAT2_mergeInputRecordsToOutput` in `starchcat.c` to check string remainders.

**Comment by alexpreynolds** [comment]
Added patches in commit 4030a4df4a7ad1bb1cac1c20a5e0a6180eccf429 to test sort order for `starch` input and to write sorted output from `starchcat` per `sort-bed` ordering.

**Comment by sjneph** [comment]
I believe that bedops -u, sort-bed's mergeSort function, and changes needed for bed_check_iterator were all addressed in changes d05d865dee3332733b3e64f6d4666bc249e64caa and f27137aadf63fcd15af76240f2dd0cbeb2a73dff though more testing would be nice before release.

**Comment by sjneph** [comment]
These changes have been released in v2p4p22.
